### DISCUSSION TITLE
Transcriptify ——The whisper that automatically compresses audio files

### DISCUSSION THREAD

**Discussion by victorGPT** [openai/whisper#1232] [Show and tell]
Hello everyone, whisper has been a great help to my work. I use a MacBook Air with an M2 processor. Using openai to process audio files is obviously a more convenient and efficient choice than processing them locally. 

However, as stated in the documentation, openai only supports audio files up to 25MB, and it is recommended by the official website that for large files, they should be split into several smaller ones before batch processing. But I have a cooler solution! - I wrote a python script called [Transcriptify](https://github.com/victorGPT/Transcriptify)

[Transcriptify](https://github.com/victorGPT/Transcriptify) uses ffmpeg to compress the sound quality before transcribing it with openai. The volume of compressed audio within about an hour and a half is all under 25MB, enough for more scenarios. 

I hope this script can be helpful to everyone. By the way, this script was programmed under the guidance of ChatGPT. Just a week ago, I was a layman who couldn't write a single line of code. Thanks to ChatPGT for changing me.

**Comment by carbium** [comment]
Great Job!

**Reply by dgoryeo** [reply]
Hi @victorGPT , is the code system dependent --linux only? I get an error in Win11:

```

ffmpeg_path = subprocess.check_output(['which', 'ffmpeg']).decode().strip()
FileNotFoundError: [WinError 2] The system cannot find the file specified

```

Thanks

**Reply by victorGPT** [reply]
Sorry, I use a Mac computer myself, so I have no problem exercising. I have noticed this issue and will fix it as soon as possible!

**Comment by landemou** [comment]
Hi @victorGPT, I currently have a professional project where I use ffmpeg-wasm to reduce file size before whisper processing. I target the aac format for a bitrate of 64 kbps. Is your conversion to 16k mp3 sufficient to guarantee the quality of the audio?
Good idea your script 👍

**Reply by victorGPT** [reply]
Friend I think you do not need to worry, I have tested ten hours of audio files, the transcription effect and the original file no difference.

**Reply by landemou** [reply]
great i will try your settings

**Reply by victorGPT** [reply]
Sorry, some of my friends gave me feedback that 32k is the best choice for balancing compression efficiency and recognition accuracy, so I suggest you change it back to 32k.

**Reply by landemou** [reply]
I also did some tests this morning and found the same thing 👍
